### PR TITLE
Github Actions migration

### LINKED ISSUE #147: Moving to Github Actions?
Since Travis has recently changed it's [pricing model](https://blog.travis-ci.com/2020-11-02-travis-ci-new-billing) and has really poor performance since quite some time, we should consider moving to Github Action.

### PR DESCRIPTION
**Pull request by offa**
CI build ported to Github Actions and Travis CI removed.

I have dropped some older compiler and upgraded the image to latest Ubuntu LTS. Please let me know if we still need older builds.

The readme shield should become visible after the first build on `master`.

Fixes #147

### CODE REVIEW

**Comment by offa** [comment]
Example (build of the fork): https://github.com/offa/seasocks/actions/runs/396638401

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+name: ci
```
Comment (mattgodbolt) [inline .github/workflows/ci.yml:1]: Oh this is so much nicer :) Thanks!

**Review by mattgodbolt — APPROVED** [review]
Looks great! Thanks so much!

File: .github/workflows/ci.yml (hunk 1 above)
Comment (offa) [reply .github/workflows/ci.yml:1]: More important, it should be notably faster! :-)

**Comment by offa** [comment]
Please let me know if there are any issues left.

**Comment by mattgodbolt** [comment]
Thanks so much :) looking great so far!

**Pull request merged**